ingenialink/ethercat/network.py

```diff
@@ -128,17 +153,23 @@ def connect_to_slave(  # type: ignore [override]
         Raises:
             ValueError: If the slave ID is not valid.
             ILError: If no slaves are found.
+            ILStateError: If slave can not reach PreOp state
 
         """
         if not isinstance(slave_id, int) or slave_id < 0:
             raise ValueError("Invalid slave ID value")
-        slaves = self.scan_slaves()
-        if len(slaves) == 0:
+        if not self.__is_master_running:
+            self._start_master()
+        if slave_id not in self.__last_init_nodes:
+            self.__init_nodes()
```

Comment:
For every connect, all the other previous nodes are set to PreOp?

Reply:
and the list of servos cleared?

Reply:
No, it is only done when the target slave id was not in the last init, in other words, when a new node was added physically or a wrong slave id is used.

List of servos is not cleared, the list cleared is the list of slaves of pysoem master class.